packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -0,0 +1,324 @@
+import { IUser } from "@esri/arcgis-rest-types";
+import { getFamilyTypes } from "../content";
+import { HubFamily } from "../types";
+import { EntityType, IHubCatalog, IHubCollection } from "./types";
+
+/**
+ * This is used to determin what IHubCatalog definition JSON object
+ * can be created
+ */
+export type WellKnownCatalog =
+  | "myContent"
+  | "myGroup"
+  | "favorites"
+  | "organization"
+  | "world";
+
+/**
+ * This is used to determin what IHubCollection definition JSON object
+ * can be created. We use HubFamily here to define most of the collections
+ * to ensure consistency
+ */
+export type WellKnownCollection =
+  | Exclude<HubFamily, "app" | "map">
+  | "appAndMap"
+  | "solution";
+
+/**
+ * A list of optional arguments to pass into getWellKnownCatalog
+ * user is the owner of the entity
+ * collectionNames is a list of names of the collections, only those collections
+ * in the catalog will be returned
+ */
+interface IGetWellKnownCatalogOptions {
+  user?: IUser;
+  collectionNames?: WellKnownCollection[];
+}
+
+/**
+ * Check if i18nScope is defined, if so add a "." at the end
+ * @param i18nScope
+ * @returns i18nScope with a "." at the end if it is defined so it stays
+ * as a valid i18nScope
+ * e.g. "project.edit.catalog" vs "project.editcatalog"
+ */
+function validateI18nScope(i18nScope: string): string {
+  return i18nScope ? `${i18nScope}.` : "";
+}
+
+/** Get a single catalog based on the name, entity type and optional requests
```

Comment:
Please run `npm run docs:serve` and verify that this comment will be included into the generated docs... I'm not sure how sensitive the generator is to the exact format of the comments

Reply:
Confirmed it's shown in the doc

![image](https://user-images.githubusercontent.com/16672774/217629007-faac3582-69cb-45c5-a807-3765bd8a7830.png)